### DISCUSSION TITLE
Usage of io.vertx.core.json.JsonObject vs jakarta.json.JsonObject ?

### DISCUSSION THREAD

**Discussion by StephenOTT** [quarkusio/quarkus#39173] [Q&A]
hi

Seeing varying usage of Jakarta vs the vertx implementations of JsonObject (and supporting json classes):

Which is the preferred version, and why?

I see for example that many quarkus tests that validate RestEasy(Reactive) are running against the vertx implementation.

Thanks!

**Comment by StephenOTT** [comment]
following up on this.

Thanks!

**Comment by gsmet** [comment]
I wouldn't say one is better than the other.

The Jakarta one is part of the JSON-P API, which is for instance used if you use JSON-B (and probably some other libraries).

The Vert.x object will be serialized with Jackson and as it comes with Vert.x which is present in most Quarkus projects, it's more or less free.

If you don't need to have JSON-P around for other things and you already have Vert.x around, I suppose I would go with the Vert.x one.

**Reply by cescoffier** [reply]
To add to what @gsmet said, we also handle Vert.x JSON Object / JSON Array directly in various places (reactive routes, event bus, redis, reactive SQL drivers...). Note that it does not need Jackson (only for mapping capabilities)
